test/integration/worker/entity.js

```diff
@@ -140,6 +140,47 @@ describe('worker: entity', () => {
       errorEvent.isEmpty().should.be.true();
     }));
 
+    it('should not make an entity when reprocessing an edited submission', testService(async (service, container) => {
+      await service.login('alice', (asAlice) =>
+        asAlice.post('/v1/projects/1/forms?publish=true')
+          .send(testData.forms.simpleEntity)
+          .set('Content-Type', 'application/xml')
+          .expect(200)
+          .then(() => asAlice.post('/v1/projects/1/forms/simpleEntity/submissions')
+            .send(testData.instances.simpleEntity.one)
+            .set('Content-Type', 'application/xml')
+            .expect(200))
+          .then(() => asAlice.patch('/v1/projects/1/forms/simpleEntity/submissions/one')
+            .send({ reviewState: 'approved' })
+            .expect(200)));
```

Comment:
I just merged a PR that allows you to call `service.login()` without passing it a callback, which I think will reduce nesting. So this could be something like:

```js
const asAlice = await service.login('alice');
await asAlice.post('/v1/projects/1/forms?publish=true')
  .send(testData.forms.simpleEntity)
  .set('Content-Type', 'application/xml')
  .expect(200);
await asAlice.post('/v1/projects/1/forms/simpleEntity/submissions')
  .send(testData.instances.simpleEntity.one)
  .set('Content-Type', 'application/xml')
  .expect(200);
await asAlice.patch('/v1/projects/1/forms/simpleEntity/submissions/one')
  .send({ reviewState: 'approved' })
  .expect(200);
```

No need to change anything of course, just wanted to put the new possibility on your radar!

Reply:
oh, thanks! i'll use it.